File: src/frontend/src/app/components/financeiro/plano-contas/model/plano-contas-grid-dto.ts
export class PlanoContasGridDTO {
  id: string;
  codigo: string;
  descricao: string;
  tipo: string;
  planoPaiCodigo?: string;
  planoPaiDescricao?: string;
  unidadeNegocioCodigo?: string;
  ativo: boolean;
  analitico: boolean;
  deleted?: boolean;

  constructor(
    id: string,
    codigo: string,
    descricao: string,
    tipo: string,
    planoPaiCodigo: string | undefined,
    planoPaiDescricao: string | undefined,
    unidadeNegocioCodigo: string | undefined,
    ativo: boolean,
    analitico: boolean,
    deleted?: boolean
  ) {
    this.id = id;
    this.codigo = codigo;
    this.descricao = descricao;
    this.tipo = tipo;
    this.planoPaiCodigo = planoPaiCodigo;
    this.planoPaiDescricao = planoPaiDescricao;
    this.unidadeNegocioCodigo = unidadeNegocioCodigo;
    this.ativo = ativo;
    this.analitico = analitico;
    this.deleted = deleted;
  }
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/financeiro/planocontas/PlanoContasGridDTO.java
package br.com.grupopipa.gestaointegrada.financeiro.planocontas;

import br.com.grupopipa.gestaointegrada.core.dto.GridDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.util.UUID;

@Builder
@Getter
@Setter
public class PlanoContasGridDTO implements GridDTO {
    private UUID id;
    private String codigo;
    private String descricao;
    private String tipo;
    private String planoPaiCodigo;
    private String planoPaiDescricao;
    private String unidadeNegocioCodigo;
    private Boolean ativo;
    private Boolean analitico;
    private Boolean deleted;
}
